spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaStreamsDefaultConfiguration.java

```diff
@@ -54,10 +63,13 @@
 
 	@Bean(name = DEFAULT_STREAMS_BUILDER_BEAN_NAME)
 	public StreamsBuilderFactoryBean defaultKafkaStreamsBuilder(
-			@Qualifier(DEFAULT_STREAMS_CONFIG_BEAN_NAME) ObjectProvider<StreamsConfig> streamsConfigProvider) {
+			@Qualifier(DEFAULT_STREAMS_CONFIG_BEAN_NAME) ObjectProvider<StreamsConfig> streamsConfigProvider,
+			Optional<List<KafkaStreamsCustomizer>> kafkaStreamsCustomizers) {
```

Comment:
Do we really need here an `Optional` wrapper? 
Won't we have an empty `List` if there is no `KafkaStreamsCustomizer` beans in the context?

On the other hand I don't think that this is good approach if we are going to manage several `KafkaStreams` in the application. This way you are going to consume all the `KafkaStreamsCustomizer`s where you might need just only one.

I wouldn't allow to inject a `List<KafkaStreamsCustomizer>` at all. A single `KafkaStreamsCustomizer` shouldn't be fully enough. And here, on the `@EnableKafkaStreams` level we would use really some conventional bean name do not clash with any other possible `KafkaStreamsCustomizer` in the application context.

Reply:
Completely right, fixing.